modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -319,6 +326,47 @@ public long getSelectedRepositoryId() {
 		return _selectedRepositoryId;
 	}
 
+	public PortletURL getSelectFolderURL(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		ItemSelector itemSelector =
+			(ItemSelector)httpServletRequest.getAttribute(
+				ItemSelector.class.getName());
+
+		FolderItemSelectorCriterion folderItemSelectorCriterion =
+			new FolderItemSelectorCriterion();
+
+		folderItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+			new FolderItemSelectorReturnType());
+		folderItemSelectorCriterion.setFolderId(getRootFolderId());
+		folderItemSelectorCriterion.setIgnoreRootFolder(true);
+		folderItemSelectorCriterion.setRepositoryId(getSelectedRepositoryId());
+		folderItemSelectorCriterion.setSelectedFolderId(getRootFolderId());
+		folderItemSelectorCriterion.setSelectedRepositoryId(
+			getSelectedRepositoryId());
+		folderItemSelectorCriterion.setShowGroupSelector(true);
+		folderItemSelectorCriterion.setShowMountFolder(false);
+
+		PortletDisplay portletDisplay = _themeDisplay.getPortletDisplay();
+
+		Repository repository = RepositoryLocalServiceUtil.fetchRepository(
+			getSelectedRepositoryId());
+
+		long groupId = getSelectedRepositoryId();
+
+		if (repository != null) {
+			groupId = _folder.getGroupId();
+		}
```

Comment:
What happens if _folder is null? You should get the groupId from the repository instead.